### PR TITLE
[test] Adding health checking integration tests - tcp

### PR DESCRIPTION
**Pull request by adisuissa**
Signed-off-by: Adi Suissa-Peleg <adip@google.com>

Commit Message: Adding health checking integration tests - tcp health checks.

Additional Description:
Adding integration tests for tcp active health checking.
This is partially to address #12286.

Risk Level: Low (tests only)
Testing: New tests.
Docs Changes: None.
Release Notes: None.

### CODE REVIEW

**Comment by adisuissa** [comment]
/retest
for windows and macos failed tests

File: test/integration/health_check_integration_test.cc [hunk 1]
```diff
@@ -258,5 +258,96 @@ TEST_P(HttpHealthCheckIntegrationTest, SingleEndpointTimeoutHttp) {
...
+
+  AssertionResult result = clusters_[cluster_idx].host_fake_raw_connection_->write("Pong");
+  RELEASE_ASSERT(result, result.message());
+
+  test_server_->waitForCounterGe("cluster.cluster_1.health_check.success", 1);
+  EXPECT_EQ(1, test_server_->counter("cluster.cluster_1.health_check.success")->value());
+  EXPECT_EQ(0, test_server_->counter("cluster.cluster_1.health_check.failure")->value());
+
+  // Clean up connections.
+  cleanupHostConnections();
```
Comment (asraa) [inline test/integration/health_check_integration_test.cc:308]: should this be in TearDown? (and for HTTP, now that I look)?

**Review by asraa — COMMENTED** [review]
Nice, thank you Adi!!! Looks perfect :)
/wait

File: test/integration/health_check_integration_test.cc (hunk 1 above)
Comment (adisuissa) [reply test/integration/health_check_integration_test.cc]: I agree. fixed.

**Comment by adisuissa** [comment]
@asraa, addressed your comments, PTAL.

**Review by asraa — APPROVED** [review]
Thanks! LGTM

**Pull request merged**